source/Calamari.Common/Features/Scripting/ScriptEngine.cs

```diff
@@ -66,6 +67,7 @@ public ScriptSyntax[] GetSupportedTypes()
         /// </returns>
         IScriptWrapper BuildWrapperChain(ScriptSyntax scriptSyntax, IVariables variables)
         {
+            bool.TryParse(variables.Get(ScriptVariables.UseDotnetScript, "false"), out bool useDotnetScript);
```

Comment:
This is the core of the change. We're defaulting to ScriptCS for backward compatibility when on prem upgrades and the variable will be set based on an environment feature toggle, which can be over ridden by a project-specific variable.